### PR TITLE
Tizen studio: profiling tool updates

### PR DESCRIPTION
**Pull request by dkson95**
### Change Description ###

Native profiling tool updates with `Address Sanitizer` and `Leak Sanitizer`

### CODE REVIEW

**Review by sangwook96 — APPROVED** [review]
LGTM 이용

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+The Leak Sanitizer is a profiling tool used to detect runtime memory leaks in Tizen native applications. With it, you can discover whether specific parts of code can potentially cause memory leak at runtime. You can also avoid leaks in memory after application execution. The Leak Sanitizer tool detects memory leak at runtime by instrumenting the code during the application compilation. A program with no bugs does not crash when the Leak Sanitizer tool is used, suggesting that the code is safe from potential memory leak.
+
+## Using the Leak Sanitizer
+
+> **Note**
+>
+> Before you run the Leak Sanitizer:
+>
+> - Make sure that you have an emulator or a connected target device running.
+> - If you want to try out the tool and do not have an applicable project to test, create a test project with the Project Wizard using a template or sample. For more information on creating a project, see [Creating Your First Tizen Mobile Native Application](../../native/get-started/mobile/first-app.md).
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:12]:
```suggestion
> [!NOTE]
> Before you run the Leak Sanitizer:
> - Make sure that you have an emulator or a connected target device running.
> - If you want to try out the tool and do not have an applicable project to test, create a test project with the Project Wizard using a template or sample. For more information on creating a project, see [Creating Your First Tizen Mobile Native Application](../../native/get-started/mobile/first-app.md).

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   Unlike Address Sanitizer, Tizen Studio checks and shows the profiling result at the application exit, as illustrated in the following figure.
+
+   In the example shown in the figure, the application crashes after `main()` function is exited.
+
+   ![Address Sanitizer output](./media/leak_sanitizer_output.png)
+
+
+## Related information
+* Dependencies
+   - Tizen Studio 1.1 and Higher
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:33]:
```suggestion
## Related information
- Dependencies
   - Tizen Studio 1.1 and Higher

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+>
+> - Make sure that you have an emulator or a connected target device running.
+> - If you want to try out the tool and do not have an applicable project to test, create a test project with the Project Wizard using a template or sample. For more information on creating a project, see [Creating Your First Tizen Mobile Native Application](../../native/get-started/mobile/first-app.md).
+
+To use the Leak Sanitizer:
+
+1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Leak Sanitizer**.
+
+   The profiling instruments the code and compiles the project for the Leak Sanitizer. Finally, it launches the application on the connected target or emulator.
+
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:19]: - Is it profiler ? or profiling?
- What is the intended message that you wish to convey can you please elaborate here. 

The profiling instruments the code and compiles the project for the Leak Sanitizer. Finally, it launches the application on the connected target or emulator.

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   ![Profiling the application](./media/leak_sanitizer_profile.png)
+
+2. After the application is launched, run the application scenario you want to test.
+
+   Unlike Address Sanitizer, Tizen Studio checks and shows the profiling result at the application exit, as illustrated in the following figure.
+
+   In the example shown in the figure, the application crashes after `main()` function is exited.
+
+   ![Address Sanitizer output](./media/leak_sanitizer_output.png)
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:28]:
```suggestion
2. After the application launches, run the application scenario that you want to test. Unlike Address Sanitizer, Tizen Studio checks and shows the profiling result when you exit the application, as illustrated in the following figure.

   In the example shown in the following figure, the application crashes after `main()` exits:

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Detecting Runtime Memory Leaks with Leak Sanitizer
+
+The Leak Sanitizer is a profiling tool used to detect runtime memory leaks in Tizen native applications. With it, you can discover whether specific parts of code can potentially cause memory leak at runtime. You can also avoid leaks in memory after application execution. The Leak Sanitizer tool detects memory leak at runtime by instrumenting the code during the application compilation. A program with no bugs does not crash when the Leak Sanitizer tool is used, suggesting that the code is safe from potential memory leak.
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:3]:
```suggestion
The Leak Sanitizer is a profiling tool used to detect runtime memory leaks in Tizen native applications. The Leak Sanitizer lets you discover whether specific parts of code can potentially cause memory leaks at runtime. You can also avoid leaks in memory after application execution. The Leak Sanitizer tool detects memory leaks at runtime by instrumenting the code during the application compilation. A program with no bugs does not crash when the Leak Sanitizer tool is used, suggesting that the code is safe from potential memory leaks.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@dkson95 Request you to kindly incorporate the comments & clear my doubt.

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md (hunk 3 above)
Comment (dkson95) [reply docs/application/tizen-studio/native-tools/leak-sanitizer.md:17]: > * Is it profiler ? or profiling?
> * What is the intended message that you wish to convey can you please elaborate here.
> 
> The profiling instruments the code and compiles the project for the Leak Sanitizer. Finally, it launches the application on the connected target or emulator.

Okay, let me fix the description here. 
```
The profiler will link your program against a runtime library containing just the bare necessities required for LeakSanitizer to work by "-fsanitize=leak" option. No compile-time instrumentation will be applied. Finally, it launches the application on the connected target or emulator to detect runtime memory leaks.
```
FYI, origin wiki page is available on https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer .

Thank you!

Comment (nawab-samsung) [reply docs/application/tizen-studio/native-tools/leak-sanitizer.md:17]: @dkson95 How about this?  (please check for technical accuracy)
```suggestion
The profiler links your program to a runtime library containing the bare necessities required for LeakSanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied.  Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@dkson95 Please have a look at the suggestion & provide your suggestion too.

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 6]
```diff
@@ -11,10 +11,7 @@ The Leak Sanitizer is a profiling tool used to detect runtime memory leaks in Ti
 
 To use the Leak Sanitizer:
 
-1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Leak Sanitizer**.
-
-   The profiling instruments the code and compiles the project for the Leak Sanitizer. Finally, it launches the application on the connected target or emulator.
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md]:
```suggestion
1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Leak Sanitizer**.
  The profiler links your program to a runtime library containing the bare necessities required for LeakSanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied.  Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks:

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@dkson95 
My bad, we have missed step 1, re-introducing it. Please incorporate the same.

**Comment by dkson95** [comment]
> checkout the webpage https://docs1.stg.tizen.org/staging/1447

@nawab-samsung thank you for this review :) and would you confirm to be merged ? Hoping this can be last updates :)

File: docs/application/tizen-studio/native-tools/address-sanitizer.md [hunk 7]
```diff
@@ -20,7 +20,7 @@ The Address Sanitizer tool can detect the following types of bugs:
 
 To use the Address Sanitizer:
 
-1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Address Sanitizer**.
+1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Memory Errors**.
 
    The profiling instruments the code and compiles the project for the Address Sanitizer. Finally, it launches the application on the connected target or emulator.
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/address-sanitizer.md:26]: I think the image needs to be changed. 
SharedScreenshot

Comment (nawab-samsung) [reply docs/application/tizen-studio/native-tools/address-sanitizer.md:26]: https://docs1.stg.tizen.org/staging/1447/application/tizen-studio/native-tools/address-sanitizer/
The latest change seems to be incorrect even if we change the image. Please verify. 

Existing document: 
https://docs.tizen.org/application/tizen-studio/native-tools/address-sanitizer/

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+> [!NOTE]
+> Before you run the Leak Sanitizer:
+> - Make sure that you have an emulator or a connected target device running.
+> - If you want to try out the tool and do not have an applicable project to test, create a test project with the Project Wizard using a template or sample. For more information on creating a project, see [Creating Your First Tizen Mobile Native Application](../../native/get-started/mobile/first-app.md).
+
+To use the Leak Sanitizer:
+
+1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Leak Sanitizer**.
+
+   The profiler links your program to a runtime library containing the bare necessities required for LeakSanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied.  Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks.
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:16]:
```suggestion
   The profiler links your program to a runtime library containing the bare necessities required for LeakSanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied.  Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks:

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@dkson95  Can you please clear my doubt. Also incorporate missed comments.

File: docs/application/tizen-studio/native-tools/address-sanitizer.md (hunk 7 above)
Comment (dkson95) [reply docs/application/tizen-studio/native-tools/address-sanitizer.md:26]: Nice catch! let me update.
Thank you!

File: docs/application/tizen-studio/native-tools/address-sanitizer.md [hunk 9]
```diff
@@ -20,7 +20,7 @@ The Address Sanitizer tool can detect the following types of bugs:
 
 To use the Address Sanitizer:
 
-1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Address Sanitizer**.
+1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Memory Errors**.
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/address-sanitizer.md:23]: The wording must match the UI text, that's why the comment: 
```suggestion
1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile For Memory Errors**.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@dkson95  All the content looks fine, please incorporate the last comment. 
Thanks

**Comment by dkson95** [comment]
@naweb

> @dkson95 All the content looks fine, please incorporate the last comment.
> Thanks

@nawab-samsung done! thank you for your valuable review with suggestion :)

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+> [!NOTE]
+> Before you run the Leak Sanitizer:
+> - Make sure that you have an emulator or a connected target device running.
+> - If you want to try out the tool and do not have an applicable project to test, create a test project with the Project Wizard using a template or sample. For more information on creating a project, see [Creating Your First Tizen Mobile Native Application](../../native/get-started/mobile/first-app.md).
+
+To use the Leak Sanitizer:
+
+1. In the **Project Explorer** view, right-click the project and select **Profile As > Profile With Leak Sanitizer**.
+
+   The profiler links your program to a runtime library containing the bare necessities required for LeakSanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied. Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks:
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:16]: Sorry but you have missed a space here: 
```suggestion
   The profiler links your program to a runtime library containing the bare necessities required for Leak Sanitizer to work by setting option `-fsanitize` to `leak`,  and no compile-time instrumentation is applied. Whilst the option is set, the application launches on the connected target or emulator to detect runtime memory leaks:

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# Detecting Runtime Memory Leaks with Leak Sanitizer
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:1]: Can we use a Sentence case for all pages in this PR?
https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
```suggestion
# Detect runtime memory leaks with Leak Sanitizer

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+The Leak Sanitizer is a profiling tool used to detect runtime memory leaks in Tizen native applications. The Leak Sanitizer lets you discover whether specific parts of code can potentially cause memory leaks at runtime. You can also avoid leaks in memory after application execution. The Leak Sanitizer tool detects memory leaks at runtime by instrumenting the code during the application compilation. A program with no bugs does not crash when the Leak Sanitizer tool is used, suggesting that the code is safe from potential memory leaks.
+
+## Using the Leak Sanitizer
```
Comment (nawab-samsung) [inline docs/application/tizen-studio/native-tools/leak-sanitizer.md:5]:
```suggestion
## Use Leak Sanitizer

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
The content in the file is fine, just a space is missed. Further I requested for changing to sentence case from title case. It would be good if you can do that,.

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md (hunk 10 above)
Comment (dkson95) [reply docs/application/tizen-studio/native-tools/leak-sanitizer.md:16]: Agreed, just LeakSanitizer is for development notation: https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer#using-leaksanitizer

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md (hunk 11 above)
Comment (dkson95) [reply docs/application/tizen-studio/native-tools/leak-sanitizer.md:1]: sure, thank you!

File: docs/application/tizen-studio/native-tools/leak-sanitizer.md (hunk 12 above)
Comment (dkson95) [reply docs/application/tizen-studio/native-tools/leak-sanitizer.md:5]: Agreed 👍

**Review by nawab-samsung — APPROVED** [review]
@dkson95 Thank you for incorporating the comments.

**Comment by dkson95** [comment]
> @dkson95 Thank you for incorporating the comments.

@nawab-samsung thank you for the approval 👍

**Pull request merged**